web/libs/core/src/pages/AccountSettings/sections/PersonalInfo.tsx

```diff
@@ -1,90 +1,121 @@
-import { useCallback, useEffect, useRef, useState } from "react";
+import { type FormEventHandler, useCallback, useEffect, useRef, useState } from "react";
 import clsx from "clsx";
-import { InputFile, useToast } from "@humansignal/ui";
+import { InputFile, ToastType, useToast } from "@humansignal/ui";
+import { API, useAPI } from "apps/labelstudio/src/providers/ApiProvider";
+import styles from "../AccountSettings.module.scss";
+import { useCurrentUserAtom } from "@humansignal/core/lib/hooks/useCurrentUser";
+import { atomWithMutation } from "jotai-tanstack-query";
+import { useAtomValue } from "jotai";
+import type { APIUser } from "@humansignal/core/types/user";
+
+/**
+ * FIXME: This is legacy imports. We're not supposed to use such statements
+ * each one of these eventually has to be migrated to core or ui
+ */
 import { Input } from "/apps/labelstudio/src/components/Form/Elements";
 import { Userpic } from "/apps/labelstudio/src/components/Userpic/Userpic";
-import { useCurrentUser } from "/apps/labelstudio/src/providers/CurrentUser";
 import { Button } from "/apps/labelstudio/src/components/Button/Button";
-import { useAPI } from "apps/labelstudio/src/providers/ApiProvider";
-import styles from "../AccountSettings.module.scss";
 
-export const PersonalInfo = () => {
-  const api = useAPI();
-  const toast = useToast();
-  const { user, fetch, isInProgress: userInProgress } = useCurrentUser();
-  const [fname, setFName] = useState("");
-  const [lname, setLName] = useState("");
-  const [email, setEmail] = useState("");
-  const [phone, setPhone] = useState("");
-  const [isInProgress, setIsInProgress] = useState(false);
-  const userInfoForm = useRef();
-  const userAvatarForm = useRef();
-  const avatarRef = useRef();
-  const fileChangeHandler = () => userAvatarForm.current.requestSubmit();
-  const avatarFormSubmitHandler = useCallback(
-    async (e, isDelete = false) => {
-      e.preventDefault();
-      const response = await api.callApi(isDelete ? "deleteUserAvatar" : "updateUserAvatar", {
-        params: {
-          pk: user?.id,
-        },
-        body: {
-          avatar: avatarRef.current.files[0],
-        },
+const updateUserAtom = atomWithMutation(() => ({
```

Comment:
What is the difference between this atom and the one over here: https://github.com/HumanSignal/label-studio/pull/7106/files#diff-4687f24e5860c45a41d1491638ed99b13d9308325f8409cde26c2ff1b53a2601R14